server/resource_server/service/service.go

```diff
@@ -264,3 +265,18 @@ func (s *service) UpdateUserPassword(username string, newPassword string, salt s
 	hashedPassword := utils.SaltAndHashPassword(newPassword, salt)
 	return s.repository.UpdateUserPassword(username, hashedPassword)
 }
+
+func (s *service) RegisterUserPrecheck(req dto.RegisterUserPrecheckDTO, iterCount int) (models.RegisterUserPrecheckResponseOutput, error) {
```

Comment:
It would be better to make this function return just an error, since the goal of the function is to register a user, the output should be either a success or a failure value, which we can express in the ``error`` type. Adding the ``RegisterUserPrecheckResponseOutput`` generation breaks the Single Responsibility principle (https://en.wikipedia.org/wiki/Single-responsibility_principle) since this method would be able to get modified for two reasons: 
1. changing business logic to register a user;
3. changing format of the response.

Thus, it's better to extract logic related to generating ``RegisterUserPrecheckResponseOutput`` into controller.go and make this handler just return an error type.